app/javascript/src/components/Common/TabsHeadersTable/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from "react";
+
+import NewNotePane from "components/Dashboard/Notes/NewNotePane";
+
+import Tabs from "./Tabs";
+import TopHeader from "./TopHeader";
+
+const TabsHeadersTable = ({ tabs, segments, tags, itemName, createTable }) => {
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+  const [showNewNotePane, setShowNewNotePane] = useState(false);
+
+  return (
+    <div className="flex w-full">
+      <Tabs
+        setIsSearchCollapsed={setIsSearchCollapsed}
+        isSearchCollapsed={isSearchCollapsed}
+        tabs={tabs}
+        segments={segments}
+        tags={tags}
+        title={itemName}
+      />
+      <div className="flex-auto">
+        <TopHeader
+          itemName={itemName}
+          setShowNewNotePane={setShowNewNotePane}
+        />
+
+        {createTable()}
+
+        <NewNotePane
+          showPane={showNewNotePane}
+          setShowPane={setShowNewNotePane}
+          fetchNotes={() => {}}
+        />
+      </div>
+    </div>
+  );
+};
```

Comment:
Out of curiosity, why do we need this component inside of `Common`?

Reply:
The `TabsHeadersTable` file contains 4 components- Tabs, Header, Table, and NewNotePane. All of them are common in `Contacts` and `Notes`, except for the `Table`. We are passing a `createTable` method that will show `NotesTable` or `ContactsTable` depending on the component provided.

Reply:
In that case, the file is symbolically a layout. We should be naming it better.